cirq/linalg/decompositions.py

```diff
@@ -992,3 +1000,49 @@ def _canonicalize_kak_vector(k_vec: np.ndarray, atol: float) -> np.ndarray:
     k_vec[need_diff, 2] *= -1
 
     return k_vec
+
+
+def num_two_qubit_gates_required(u: np.ndarray, atol: float = 1e-8) -> int:
```

Comment:
I'd keep CNOT or CZ in the name since if one were to compute the minimum number of gates of a different type required to implement a given unitary one would generally obtain a different result(*).

Also, we may in future consider decompositions into other gates, so a name containing CNOT or CZ is future-proof.

---
(*) For example, SWAP requires 3 CNOTs/CZs, but (trivially) only 1 SWAP.

Reply:
(SWAP is not universal, so it's a poor example, but the larger point remains.)